packages/eas-cli/src/uploads.ts

```diff
@@ -57,26 +57,30 @@ async function obtainS3PresignedPostAsync(
 export async function uploadWithPresignedPostAsync(
   stream: Readable | Buffer,
   presignedPost: PresignedPost,
-  handleProgressEvent: ProgressHandler
-) {
+  handleProgressEvent?: ProgressHandler
+): Promise<string> {
   const form = new FormData();
   for (const [fieldKey, fieldValue] of Object.entries(presignedPost.fields)) {
     form.append(fieldKey, fieldValue);
   }
   form.append('file', stream);
   const formHeaders = form.getHeaders();
   let uploadPromise = got.post(presignedPost.url, { body: form, headers: { ...formHeaders } });
+
   if (handleProgressEvent) {
     uploadPromise = uploadPromise.on('uploadProgress', progress =>
       handleProgressEvent({ progress })
     );
+    try {
+      const response = await uploadPromise;
+      handleProgressEvent({ isComplete: true });
+      return String(response.headers.location);
+    } catch (error) {
+      handleProgressEvent({ isComplete: true, error });
```

Comment:
Not sure if we need these synthetic events after success/error because the caller can as easily trigger the necessary actions after the promise has resolved. Removing them (and additionally using the `Progress` event from got without wrapping it) would also greatly simplify this code:
```ts
const uploadPromise = got.post(presignedPost.url, { body: form, headers: { ...formHeaders } });
if (handleProgressEvent) {
  uploadPromise.on('uploadProgress', handleProgressEvent);
}
const response = await uploadPromise;
return String(response.headers.location);
```

Reply:
This is much more concise, but looking through the code for a progress handler, it seems to expect an `isComplete` argument.

See this recent PR in particular: https://github.com/expo/eas-cli/pull/153/files#diff-54650d756e5f6628f0b3c7c2564229e97085d05dacc0fac8263e0fcb9647da47R70

Simplifying this code seems desirable, but perhaps worth consulting @EvanBacon first and doing in a separate PR.

Reply:
I think it might be still best to keep this function as simple as possible and then implement the progress handler interface in `publish.ts`, because it will need to keep track of all the uploads in order to show a combined progress bar. And I think the progress bar can be added in another PR.